statistics/handle/ddl.go

```diff
@@ -44,21 +46,91 @@ func (h *Handle) HandleDDLEvent(t *util.Event) error {
 			}
 		}
 	case model.ActionAddTablePartition, model.ActionTruncateTablePartition:
-		pruneMode := h.CurrentPruneMode()
-		if pruneMode == variable.Static {
-			for _, def := range t.PartInfo.Definitions {
-				if err := h.insertTableStats2KV(t.TableInfo, def.ID); err != nil {
-					return err
-				}
+		for _, def := range t.PartInfo.Definitions {
+			if err := h.insertTableStats2KV(t.TableInfo, def.ID); err != nil {
+				return err
 			}
 		}
-		if pruneMode == variable.Dynamic {
-			// TODO: need trigger full analyze
-		}
 	case model.ActionDropTablePartition:
 		pruneMode := h.CurrentPruneMode()
-		if pruneMode == variable.Dynamic {
-			// TODO: need trigger full analyze
+		if pruneMode == variable.Dynamic && t.PartInfo != nil {
+			if err := h.updateGlobalStats(t.TableInfo); err != nil {
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+// analyzeOptionDefault saves the default values of NumBuckets and NumTopN.
+// These values will be used in dynamic mode when we drop table partition and then need to merge global-stats.
+// These values originally came from the analyzeOptionDefault structure in the planner/core/planbuilder.go file.
+var analyzeOptionDefault = map[ast.AnalyzeOptionType]uint64{
```

Comment:
How about making this structure in `planbuilder.go` exposed and using it directly.

Reply:
It seems will cause the `import cycle`